docs/pages/api.rst

```diff
@@ -1429,6 +1444,46 @@ Signature: ``Collection::permutate(int $size, $value);``
     $collection = Collection::fromIterable(['a', 'b'])
         ->permutate(); // [['a', 'b'], ['b', 'a']]
 
+pipe
+~~~~
+
+Pipe together multiple operations and apply them in succession to the collection items.
+To maintain a lazy nature, each operation needs to return a ``Generator``.
+Custom operations and operations provided in the API can be combined together.
+
+Interface: `Pipeable`_
+
+Signature: ``Collection::pipe(callable ...$callables);``
+
+.. code-block:: php
+
+    $square = static function ($collection): Generator {
+        foreach ($collection as $item) {
+            yield $item ** 2;
+        }
+    };
+
+    $toString = static function ($collection): Generator {
+        foreach ($collection as $item) {
+            yield (string) $item;
+        }
+    };
+
+    $times = new class() extends AbstractOperation {
+        public function __invoke(): Closure
+        {
+            return static function ($collection): Generator {
+                foreach ($collection as $item) {
+                    yield "{$item}x";
+                }
+            };
+        }
+    };
+
+    Collection::fromIterable(range(1, 5))
+        ->pipe($square, Reverse::of(), $toString, $times())
+        ->all(); // ['25x', '16x', '9x', '4x', '1x']
+
```

Comment:
I hope this is a good definition and showcase of `pipe`. Let me know if you think anything else should be included in the description.

Another thing I noticed: it's return type is `self` vs `CollectionInterface` like most other methods: https://github.com/loophp/collection/blob/master/src/Collection.php#L610. Is that just an error or was there a reason for it?

Reply:
This is something that I should sort one day. Returning `self` or the `interface`. I haven't found a proper way to do it yet.

Regarding the explanation, it's perfect :)